dataflow/src/main/java/org/checkerframework/dataflow/reachingdefinitions/ReachingDefinitionsTransfer.java

```diff
@@ -0,0 +1,57 @@
+package org.checkerframework.dataflow.reachingdefinitions;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.dataflow.analysis.ForwardTransferFunction;
+import org.checkerframework.dataflow.analysis.RegularTransferResult;
+import org.checkerframework.dataflow.analysis.TransferInput;
+import org.checkerframework.dataflow.analysis.TransferResult;
+import org.checkerframework.dataflow.cfg.UnderlyingAST;
+import org.checkerframework.dataflow.cfg.node.AbstractNodeVisitor;
+import org.checkerframework.dataflow.cfg.node.AssignmentNode;
+import org.checkerframework.dataflow.cfg.node.LocalVariableNode;
+import org.checkerframework.dataflow.cfg.node.Node;
+
+import java.util.List;
+
+/** A reaching definitions transfer function. */
```

Comment:
This would be a good place to document the intuition of this transfer function.

Reply:
Am I suppose to write documents for the logic of the transfer func here?

Reply:
rough idea